sidekick_core/test/sidekick_command_runner_test.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sidekick_test/fake_stdio.dart';
+import 'package:sidekick_test/sidekick_test.dart';
+import 'package:test/fake.dart';
+import 'package:test/test.dart';
+
+void main() {
+  setUp(() {
+    env['SIDEKICK_ENABLE_UPDATE_CHECK'] = 'true';
+  });
+
+  tearDown(() {
+    addTearDown(() => env['SIDEKICK_ENABLE_UPDATE_CHECK'] = null);
+  });
+
+  final expectedCliVersionIntegrityWarning =
+      'The sidekick_core version is incompatible with the bash scripts '
+      'in /tool and entrypoint because you probably updated the '
+      'sidekick_core dependency of your CLI package manually.\n'
+      'Please run ${cyan('dash sidekick update')} to repair your CLI.';
+  final expectedOutdatedWarning = '${yellow('Update available!')}\n'
+      'Run ${cyan('dash sidekick update')} to update your CLI.';
+
+  test('prints warnings to update outdated CLI', () async {
+    final fakeStderr = FakeStdoutStream();
+    await insideFakeProjectWithSidekick(
+      (tempDir) async {
+        await overrideIoStreams(
+          stderr: () => fakeStderr,
+          body: () async {
+            final runner = initializeSidekick(
+              name: 'dash',
+            );
+
+            await runner.run(['-h']);
```

Comment:
We're doing a lookup at pub.dev, right? That part should be mocked

Reply:
I made `VersionChecker` injectable for `SidekickCommandRunner` so we can fake being offline